manager/integration/tests/common.py

```diff
@@ -839,23 +843,85 @@ def check_volume_replicas(volume, spec, tag_mapping):
 # Default argument is mutable on this function, but it's fine since we're only
 # using it as an empty tag list to pass to the server and will never actually
 # modify it.
-def set_node_tags(client, node, tags=[]):  # NOQA
+def set_node_tags(client, node, tags=[], retry=False):  # NOQA
     """
     Set the tags on a node without modifying its scheduling status.
+    Retry if "too many retries error" happened.
     :param client: The Longhorn client to use in the request.
     :param node: The Node to update.
     :param tags: The tags to set on the node.
     :return: The updated Node.
     """
-    return client.update(node, allowScheduling=node.allowScheduling,
-                         tags=tags)
+    if retry is True:
+        for _ in range(NODE_UPDATE_RETRY_COUNT):
+            try:
+                node = client.update(node,
+                                     allowScheduling=node.allowScheduling,
+                                     tags=tags)
+            except Exception as e:
+                if retry_error in str(e.error.message):
+                    time.sleep(NODE_UPDATE_RETRY_INTERVAL)
+                    continue
+                print(e)
+                raise
+            else:
+                break
 
+        return node
+    else:
+        return client.update(node, allowScheduling=node.allowScheduling,
+                             tags=tags)
```

Comment:
Suggest changing to this, so don't need to have a long if-else statement. The same for other functions. 

```suggestion
    if not retry:
        return client.update(node, allowScheduling=node.allowScheduling, tags=tags)
                   
    for _ in range(NODE_UPDATE_RETRY_COUNT):
        try:
            node = client.update(node,
                                 allowScheduling=node.allowScheduling,
                                 tags=tags)
        except Exception as e:
            if retry_error in str(e.error.message):
                time.sleep(NODE_UPDATE_RETRY_INTERVAL)
                continue
            print(e)
            raise
        else:
            break

    return node
```

Reply:
Addressed.